Backend-Challenges.md

```diff
@@ -65,32 +80,116 @@ In `Challenge_6.t.sol`, we want to get the token URI from the contract given a c
+
+`vm.startPrank()` => A Foundry [cheat code](https://book.getfoundry.sh/cheatcodes/) that simulates the `msg.sender` to be the address specified. More on `msg.sender` later.
+
+`console.log()` => The print statement of Foundry. You can also use this to log variables. Initially, you won't see the output. For that, you need to supply the `-vv` flag to the test. Try running the test using `forge test --match-contract Challenge7 -vv`. Also, try adding / removing v's and see what happens.
 
 ### Mint Functions
 
 Minting is the core of getting tokens out there on the blockchain. NFTs are Non-Fungible Tokens, and they have to be minted. In the past, these functions required a lot of [gas](https://cryptomarketpool.com/gas-in-solidity-smart-contracts), but more and more standards are being developed to make minting cost less ETH, such as ERC721A.
 
-As you might have noticed, NFTs have a token id. Most of the time, when you deploy a fresh collection, the token ID starts at 0 or 1. Then, every time an NFT is minted, the token id increases and is assigned to the minted NFT.
+As you might have noticed, NFTs have a token id. Most of the time, when you deploy a fresh collection, the convention is to have the token ID start at 0 or 1. Then, every time an NFT is minted, the token id increases and is assigned to the minted NFT.
+
+## Challenge 8 - Adding some checks
+
+The goal of this challenge is to add a couple of checks to the mint function. Right now, even though we have set a max supply, it doesn't actually do anything; we can easily mint more than the max supply. So let's fix that.
+
+A way to do this, is to add an if-statement within the mint function that checks if the total supply + mint amount exceeds the max supply. If so, we revert the transaction and maybe we send out a little message so the user will know why the transaction reverted.
+
+An easier way to do this though, is by adding a `require()` statement within the function. You can read more about `require()` [here](https://www.alchemy.com/overviews/solidity-require).
+
+Take a look at `Challenge_8.t.sol`. As per usual, we have added some tests, and we need to pass those tests.
+
+- `testMintOverMax()` => A fuzz test that tries to mint over the max allowed supply
+- `testMintZero()` => A test that mints... zero NFTs
+- `testMintCA()` => A test that simulates a CA (Contract Account) minting instead of a human (EOA or Externally Owned Account)
+
+So, now let's improve the mint function to pass these checks. Our function should:
+
+- `require()` that the mint amount > 0
+- `require()` that the mints do not exceed the max allowed supply
+- `require()` that the user minting is a human and not a contract
+
+**HINT:** For the third `require()`, you can make use of `msg.sender` and `tx.origin`
```

Comment:
Id suggest a dropdown again, otherwise the hint immediately spoils it. As it is you are already hinting them quite a lot. (Also with the require)

Reply:
Will add dropdowns for all hints 👍